### ISSUE TITLE
`solve_quartic` is not scale-invariant: quartics whose roots are all small lose real roots and return non-roots, in float64 too

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4833]
`solve_quartic` compares several intermediate quantities against scales floored at 1: `y_residual_scale`, `R_sq_scale`, and (with #4669) `root_residual_scale` and `derivative_scale`. When all of a quartic's roots are small, its coefficients and terms fall far below 1. Those tests then become absolute and stop distinguishing anything. Rescaling the unknown by an exact power of two (x = t·u) does not change the roots, but it changes the answer.

## Repro (main `012997ceb`, torch 2.14.0, CPU)

```python
import torch
from kornia.geometry.solvers import solve_quartic

# (x - 1e-3)(x - 2e-3)(x^2 - 6e-3 x + 1e-5): real roots 1e-3 and 2e-3
c = torch.tensor([[1.0, -0.009, 3e-05, -4.2e-08, 2e-11]], dtype=torch.float64)
print(solve_quartic(c))                                          # [0, 0, 0.0034142, 0.0005858]
t = 2.0 ** -9
print(solve_quartic(c / t ** torch.arange(5, dtype=torch.float64)) * t)  # [0, 0, 0.002, 0.001]
```

The same polynomial at unit scale, `(x-1)(x-2)(x^2-6x+10)`, is solved correctly in both dtypes. At 1e-3 scale the returned values are `2 ± sqrt(2)` times 1e-3, the roots of a different quadratic.

## Measured

This uses 20k seeded rows per family with truth from the constructed roots. Each row is a family multiplied by s (roots × s, coefficient i × s^i). Cells are rows missing a real root / rows returning a non-root, judged at 1% (float32) or 1e-5 (float64) of `max(1, |r|/s)`:

| family (main) | s = 1 | s = 0.01 | s = 0.001 |
|---|---|---|---|
| 2 real + complex pair, float32 | 117 / 83 | 2,766 / 2,825 | 5,117 / 4,879 |
| 2 real + complex pair, float64 | 0 / 0 | 6 / 6 | 5,254 / 4,844 |
| near-square (#4474), float32 | 0 / 9,365 | 0 / 5,283 | 0 / 6,360 |
| near-square (#4474), float64 | 0 / 2,174 | 0 / 2,138 | 0 / 1,459 |

With an exact power-of-two prescale to a unit root bound (`t = 2^round(log2(max(|A|, |B|^(1/2), |C|^(1/3), |D|^(1/4))))` on the monic coefficients), every row stays near its s = 1 value at every s:

- 2-real float32 is 103-121 / 62-85.
- 2-real float64 is 0 / 0.
- near-square float64 is 0 / 2,138-2,174.

The target scale interacts with the other floors: prescaled to a unit bound, #4669's filter leaves 20-104 float32 near-square rows of 20k, where unscaled it leaves 0. The target is worth choosing with that in mind. #4669's residual filter removes the near-square non-roots at unit scale, but not at s = 0.01 in float32 or s = 0.001 in float64, for the same reason.

## Expected / suggested fix

Normalize at entry: divide coefficient i by t^i with t a power of two near the root bound, solve, and multiply the roots by t. Power-of-two scaling is exact barring overflow and underflow, so rows already at unit scale are bitwise unchanged. The gradient flows through the rescale; `t` itself can be detached. Pin the literal above in float32 and float64, and a scaled near-square row.

Found while reviewing #4669; separate from it.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
